components/EventCards.vue

```diff
@@ -0,0 +1,199 @@
+<script setup lang="ts">
```

Comment:
[Minor]
For this component, it could be worth separating the layout and logic into separate files.
But that's a tradeoff:
as the downside of split, it'll become harder to make changes that affect both logic and layout.